file/nifile.yml

```diff
@@ -0,0 +1,816 @@
+        default:
+          $ref: '#/responses/Error'
+  /v1/service-groups/Default/files/{id}/data:
+    parameters:
+      - $ref: '#/parameters/id'
+    get:
+      tags: [files]
+      summary: Download file
+      description: >-
+        Downloads a file from the Skyline File service. The file is returned in
+        a single HTTP response. The file's name metadata property is appended
+        to the response header so that the file is downloaded with the
+        appropriate file name.When the the inline query parameter is not
+        specified or is set to false, the Content-Disposition header is set to
+        'attachment' and the MIME type is set to binary so that the browser
+        handles the response as a file download.  When the inline query
+        parameter is set to true, the Content-Disposition header and MIME type
+        remain unset, and the file contents must be handled by the caller.
+      operationId: ReceiveFile
+      x-ni-operation: downloadData
+      x-ni-privilege: DownloadFiles
+      x-ni-request-variables:
+        - REQUEST_METHOD
+      parameters:
+        - in: query
+          name: inline
+          description: Whether to return the file data inline rather than as an attachment.
+          type: boolean
+          default: false
+      produces:
+        - application/octet-stream
+        - application/json
+      responses:
+        200:
+          description: OK
+          schema:
+            description: The file data
+            type: file
+          examples:
+            application/octet-stream:
+              File data
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/service-groups/Default/files/{id}/update-metadata:
+    parameters: 
+      - $ref: '#/parameters/id'
+    post:
+      tags: [files]
+      summary: Update file metadata properties
+      description: >-
+        Updates an existing file's metadata with the supplied metadata
+        properties. The ReplaceExisting value determines whether the current
+        list should be entirely replaced by the supplied list, or if entries
+        should be added to (or updated) in the existing set of file properties.
+      operationId: UpdateMetadata
+      x-ni-operation: updateMetadata
+      x-ni-privilege: ModifyFiles
+      parameters:
+        - in: body
+          name: metadata
+          description: The file's metadata and options for updating it
+          required: true
+          schema:
+            description: The file's metadata and options for updating it
+            title: UpdateMetadataRequest
+            type: object
+            properties:
+              replaceExisting:
+                description: >-
+                  Whether the current list should be entirely replaced by the
+                  supplied list, or if entries should be added to (or updated)
+                  in the existing set of file properties.
```

Comment:
Determines whether the current list should be entirely replaced by the specified list.